bluemira/equilibria/coils/_grouping.py

```diff
@@ -0,0 +1,1088 @@
+            Removed names if not at top level of nested stack
+
+        Notes
+        -----
+        If a nested coilgroup is empty it is also removed from the parent coilgroup
+
+        """
+        names = self.name
+
+        if _top_level and (remainder := set(coil_name) - set(names)):
+            raise EquilibriaError(f"Unknown coils {remainder}")
+
+        c_names = []
+        removed_names = []
+
+        for c in self._coils:
+            if isinstance(c, CoilGroup):
+                removed_names.extend(c.remove_coil(*coil_name, _top_level=False))
+            elif c.name in coil_name:
+                c_names.append(c.name)
+
+        to_remove = [names.index(c_n) for c_n in c_names]
+
+        coils = np.array(self._coils, dtype=object)
+        mask = np.full(coils.size, True, dtype=bool)
+        mask[to_remove] = False
+
+        for no, c in enumerate(self._coils):
+            if c.name == []:
+                mask[no] = False
+
+        self._coils = tuple(coils[mask])
+
+        if not _top_level:
+            return removed_names + to_remove
+
+    @classmethod
+    def from_group_vecs(cls, eqdsk: EQDSKInterface):
+        """
+        Initialises an instance of CoilSet from group vectors. This has been
+        implemented as a dict operation, because it will occur for eqdsks only.
+        Future dict instantiation methods will likely differ, hence the
+        confusing name of this method.
+        """
+        pfcoils = []
+        cscoils = []
+        passivecoils = []
+        for i in range(eqdsk.ncoil):
+            dx = eqdsk.dxc[i]
+            dz = eqdsk.dzc[i]
+            if abs(eqdsk.Ic[i]) < I_MIN:
+                # Catch CREATE's crap 0's
+                passivecoils.append(
+                    Coil(
+                        eqdsk.xc[i],
+                        eqdsk.zc[i],
+                        current=0,
+                        dx=dx,
+                        dz=dz,
+                        ctype="NONE",
+                        control=False,
+                    )
+                )
+            else:
+                if dx != dz:  # Rough and ready
+                    cscoils.append(
+                        Coil(
+                            eqdsk.xc[i],
+                            eqdsk.zc[i],
+                            current=eqdsk.Ic[i],
+                            dx=dx,
+                            dz=dz,
+                            ctype="CS",
+                        )
+                    )
+                else:
+                    coil = Coil(
+                        eqdsk.xc[i],
+                        eqdsk.zc[i],
+                        current=eqdsk.Ic[i],
+                        dx=dx,
+                        dz=dz,
+                        ctype="PF",
+                    )
+                    coil.fix_size()  # Oh ja
+                    pfcoils.append(coil)
+
+        coils = pfcoils
+        if len(cscoils) != 0:
+            coils.extend(cscoils)
```

Comment:
Is the `if` necessary here? 
```suggestion
        coils.extend(cscoils)
```